asynciterator.ts

```diff
@@ -1405,15 +1384,29 @@ export class MultiTransformIterator<S, D = S> extends TransformIterator<S, D> {
     // Remove transformers that have ended
     const transformerQueue = this._transformerQueue, optional = this._optional;
     let head, item;
-    while ((head = transformerQueue[0]) && head.transformer.done) {
+    while ((head = transformerQueue[0]) !== undefined) {
```

Comment:
I feel the changes here could do with a little explanation:

In short - since `end` is now triggered by calling `.read`, we need to make sure that we now always have triggered at least one `.read` call before checking if the iterator is `.done`.

At the same time I added the ability to add data from multiple consecutive Transformers in the queue so long as they have data ready; this can reduce excessive calls to `_read` if the transformers in the queue only produce a small number of elements each.